src/main/java/igym/services/GymService.java

```diff
@@ -73,20 +75,25 @@ public List<Gym> findAllGyms() {
         return gyms;
     }
 
+    @Transactional
     public void deleteGym(UUID id) {
-        logger.info("Attempting to inactivate gym with id {}", id);
+        logger.info("Attempting to inactivate gym with id {}", id);        
         Gym gym = findById(id);
 
-        if (gym.getStatus() == Status.inactive) {
-            logger.warn("Gym with id {} is inactive", id);
-            throw new GymNotFoundException("Gym with id " + id + " not found");
-        }
-
+        inactivateWorkouts(gym);
         gym.setStatus(Status.inactive);
         gymRepository.save(gym);
         logger.info("Gym with id {} inactivated", id);
     }
-    
+
+    private void inactivateWorkouts(Gym gym) {
+        List<Workout> workouts = workoutRepository.findByGym(gym);
+        if(workouts != null) {
+            workouts.forEach(w -> workoutService.deleteWorkout(w.getId()));
```

Comment:
This method is called deleting the gym and deleting the workout by ID. But if there are no workouts or workouts are inactive I cannot delete the gym.